helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -757,9 +762,22 @@ private void handleEvent(ClusterEvent event, BaseControllerDataProvider dataProv
     }
     event.addAttribute(AttributeName.ControllerDataProvider.name(), dataProvider);
 
-    logger.info(String.format("START: Invoking %s controller pipeline for cluster %s event: %s  %s",
-        manager.getClusterName(), dataProvider.getPipelineName(), event.getEventType(),
-        event.getEventId()));
+    String eventSessionId = event.getAttribute(AttributeName.EVENT_SESSION.name());
+    if (eventSessionId != null) {
+      String managerSessionId = manager.getSessionId();
+      if (!eventSessionId.equals(managerSessionId)) {
+        logger.warn(
+            "Controller pipeline is not invoked because cluster manager {} lost leadership for "
```

Comment:
This warn message is misleading, it does not necessarily mean the current controller is not the leader of the cluster. This just means we discarded event that was generated from an old session.

Reply:
The cluster manager's session changed, lost and then acquired leadership again. Yep, you are right. I'll make the log clearer.